### ISSUE TITLE
no_self_use fails to catch methods with empty bodies

### ISSUE THREAD

**Issue by scop** [pylint-dev/pylint#6993]
### Bug description

no_self_use does not catch methods that have empty bodies (just a docstring or a `pass`)

### Command used

With the test cases from https://github.com/PyCQA/pylint/pull/6994 added:

```shell
pytest tests/test_functional.py::test_functional[no_self_use]
```

### Pylint output

```shell
____________________________________________________ test_functional[no_self_use] ____________________________________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x7f13d49a0910>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "no_self_use":
E       
E       Expected in testdata:
E         29: no-self-use
E         34: no-self-use

pylint/testutils/lint_module_test.py:145: AssertionError
```

### Expected behavior

No errors.

### Pylint version

```shell
pylint 2.15.0-dev0
astroid 2.11.6
Python 3.10.0 (default, Oct  8 2021, 12:02:33) [GCC 9.3.0]
```

### Additional dependencies

https://github.com/PyCQA/pylint/pull/6994

**Comment by Pierre-Sassoulas** [comment]
The discussion on this issue happened in #6994, if you care about the issue please vote [here](https://github.com/PyCQA/pylint/pull/6994#issuecomment-1321117560)

**Comment by Pierre-Sassoulas** [comment]
Closing as won't do, but we'll reconsider later if a clear consensus emerge on what should denote abstractness in the python community. See discussion in linked PR for more details.
